### PR TITLE
play_tune: Write new plugin

### PR DESCRIPTION
**Pull request by mortenfyhn**
Now that PX4 has proper handling of incoming PLAY_TUNE/PLAY_TUNE_V2 mavlink messages, I've written a tiny plugin that allows sending these tunes from ROS.

I've added a mavros_msgs/PlayTuneV2 for this, to make the interface clear, which includes the tune format and tune string.

Tested on a pixhawk with PX4 v1.11, and it plays as expected.

### CODE REVIEW

File: mavros_extras/src/plugins/play_tune.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  private:
+    ros::NodeHandle nh;
+    ros::Subscriber sub;
+
+    void callback(const mavros_msgs::PlayTuneV2::ConstPtr& tune)
+    {
+        auto msg = mavlink::common::msg::PLAY_TUNE_V2{};
+        msg.target_system = m_uas->get_tgt_system();
+        msg.target_component = m_uas->get_tgt_component();
```
Comment (vooon) [inline mavros_extras/src/plugins/play_tune.cpp:31]: Please use `UAS::msg_set_target()`

https://github.com/mavlink/mavros/blob/master/mavros/include/mavros/mavros_uas.h#L387

File: mavros_extras/src/plugins/play_tune.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                break;
+
+            case mavros_msgs::PlayTuneV2::TUNE_FORMAT_MML_MODERN:
+                msg.format = static_cast<uint32_t>(mavlink::common::TUNE_FORMAT::MML_MODERN);
+                break;
+
+            default:
+                ROS_ERROR_NAMED("play_tune", "Invalid tune format: '%i'", tune->format);
+                return;
+        }
```
Comment (vooon) [inline mavros_extras/src/plugins/play_tune.cpp:46]: Usually that is not necessary, we just follow mavlink enums in msgs.

File: mavros_extras/src/plugins/play_tune.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        std::strncpy(msg.tune.data(), tune->tune.c_str(), sizeof(msg.tune));
+        msg.tune.back() = '\0';
```
Comment (vooon) [inline mavros_extras/src/plugins/play_tune.cpp:49]: Please use `mavlink::set_string_z()`.

https://github.com/ArduPilot/pymavlink/blob/b1efa7e5fd50561dfa9f8a491e4a55c8e0eedec9/generator/CPP11/include_v2.0/message.hpp#L143

**Review by vooon — CHANGES_REQUESTED** [review]
Overall plugin looks good, but few changes needed:
1. see comments
2. please follow uncrustify part of https://github.com/mavlink/mavros/blob/master/CONTRIBUTING.md

File: mavros_extras/src/plugins/play_tune.cpp (hunk 1 above)
Comment (mortenfyhn) [reply mavros_extras/src/plugins/play_tune.cpp:31]: :+1:

File: mavros_extras/src/plugins/play_tune.cpp (hunk 3 above)
Comment (mortenfyhn) [reply mavros_extras/src/plugins/play_tune.cpp:49]: Thanks, that's much better. `strncpy` works, but it sure feels bug prone.

File: mavros_extras/src/plugins/play_tune.cpp (hunk 2 above)
Comment (mortenfyhn) [reply mavros_extras/src/plugins/play_tune.cpp:46]: Yeah, it felt a bit redundant, but at the same time, repeating the enums in the message definitions means that the message sender doesn't have to depend on mavlink, just on mavros_msgs. Is that what you meant, have the sender do `msg.format = mavlink::common::TUNE_FORMAT::QBASIC1_1` and similar?

Comment (mortenfyhn) [reply mavros_extras/src/plugins/play_tune.cpp:46]: Looking at some other messages, I think I understand now. I'll update it to assign the format field directly from the mavros_msgs/PlayTuneV2 to the mavlink message. As long as the enum definition in the the mavros message matches the enum in mavlink, it's all good.

**Pull request merged**